uni/lib/view/login/login.dart

```diff
@@ -231,7 +231,58 @@ class LoginPageViewState extends State<LoginPageView> {
       Navigator.pushReplacementNamed(
           context, '/${DrawerItem.navPersonalArea.title}');
     } else if (status == RequestStatus.failed) {
-      ToastMessage.error(context, 'O login falhou');
+      final errorMessage =
+          Provider.of<SessionProvider>(context, listen: false).errorMessage;
+
+      if (errorMessage == "A palavra-passe expirou") {
+        updatePasswordDialog();
+      } else {
+        ToastMessage.error(context, (errorMessage ?? 'Erro no login'));
```

Comment:
Again, this is why you maybe should change this to an execption because `A palavra-passe expirou` might not be valid in other languages. (don't handle the exception here, handle it on a different function if the completer throws an error)